### PR TITLE
Remove underscores from YAML

### PR DESCRIPTION
**Pull request by kabir**
WIseAgentsYAMLObject deals with the underscores when (de)serializing.

There is also a hook to validate the data.

Note that since I believe WiseAgentMessage is something internal, I did not make it implement WiseAgentsYAMLObject so that still has the underscores.

### CODE REVIEW

**Review by maeste — APPROVED** [review]
I'm approving this. LGTM. I let @fjuma merge it; it is better to have also her review for this

File: src/wiseagents/schema/neo4j-langchain-wise-agent-graph-db.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      "description": "Whether to refresh the graph schema",
+      "default": true
+    },
+    "url": {
+      "type": "string",
+      "description": "The URL of the Neo4j database (the username, password, and database name to be used"
+    }
+  },
+  "required": [
+    "collection_name",
```
Comment (fjuma) [inline src/wiseagents/schema/neo4j-langchain-wise-agent-graph-db.json]: Just a note, I know this file was deleted in the next commit but whenever this is added back - properties are required too.

File: src/wiseagents/graphdb/lang_chain_wise_agent_graph_db.py [hunk 2]
```diff
@@ -97,6 +97,7 @@ class Neo4jLangChainWiseAgentGraphDB(LangChainWiseAgentGraphDB):
     and a corresponding Neo4j vector database.
     """
     yaml_tag = u'!wiseagents.graphdb.Neo4jLangChainWiseAgentGraphDB'
+    expects_underscores_in_yaml_keys = False
```
Comment (fjuma) [inline src/wiseagents/graphdb/lang_chain_wise_agent_graph_db.py:100]: This doesn't need to block this PR from being merged but just a question. Is there a reason why this is only added in this file? I don't see it being used anywhere. If it's not needed, we can remove it in a separate PR after merging.

**Review by fjuma — APPROVED** [review]
Added a small question but it doesn't need to block this from being merged. If a change is needed, it can be added in a follow-up PR. Thanks!

**Comment by maeste** [comment]
@kabir  I have a comment also for a possible follow-up. Now that I'm start implementing a new class extending WiseAgentsYAMLObject I'm wondering if we cannot set in the abstract class yaml_loader = WiseAgentsLoader instead of require all subclasses to do that

File: src/wiseagents/graphdb/lang_chain_wise_agent_graph_db.py (hunk 2 above)
Comment (kabir) [reply src/wiseagents/graphdb/lang_chain_wise_agent_graph_db.py:100]: Ah yeah, I was thinking I had to phase in the change but it worked out in another way https://github.com/wise-agents/wise-agents/pull/364

**Comment by kabir** [comment]
@maeste Definitely, I had that in mind but forgot all about it. Please either do it, or create an issue for me

**Comment by maeste** [comment]
Better if I do it as part of my next PR which basically is touching everything in core.py and wiseagents.agents

**Pull request merged**